src/Symfony/Component/Notifier/Bridge/Smsbox/SmsboxOptions.php

```diff
@@ -11,375 +11,203 @@
-    public const MESSAGE_DAYS_THURSDAY = 4;
-    public const MESSAGE_DAYS_FRIDAY = 5;
-    public const MESSAGE_DAYS_SATURDAY = 6;
-    public const MESSAGE_DAYS_SUNDAY = 7;
-
-    public const MESSAGE_UDH_6_OCTETS = 1;
-    public const MESSAGE_UDH_7_OCTETS = 2;
-    public const MESSAGE_UDH_DISABLED_CONCAT = 0;
-
-    private array $options;
-
-    public function __construct(array $options = [])
+    public function __construct(private array $options = [])
     {
-        $this->options = [];
     }
 
     public function getRecipientId(): null
     {
         return null;
     }
 
-    public function mode(string $mode)
+    public function mode(SmsboxMessageMode $mode): self
     {
-        $this->options['mode'] = self::validateMode($mode);
+        $this->options['mode'] = $mode->value;
 
         return $this;
     }
 
-    public function strategy(int $strategy)
+    public function strategy(SmsboxMessageStrategy $strategy): self
     {
-        $this->options['strategy'] = self::validateStrategy($strategy);
+        $this->options['strategy'] = $strategy->value;
 
         return $this;
     }
 
-    public function date(string $date)
-    {
-        $this->options['date'] = self::validateDate($date);
-
-        return $this;
-    }
-
-    public function hour(string $hour)
-    {
-        $this->options['heure'] = self::validateHour($hour);
-
-        return $this;
-    }
-
-    /**
-     * This method mustn't be set along with date and hour methods.
-     */
-    public function dateTime(\DateTime $dateTime)
+    public function date(string $date): self
     {
-        $this->options['dateTime'] = self::validateDateTime($dateTime);
+        if (isset($this->options['dateTime'])) {
+            throw new InvalidArgumentException(sprintf('Either %1$s::dateTime() or %1$s::date() and %1$s::hour() must be called, but not both.', self::class));
+        }
 
-        return $this;
-    }
+        if (false === \DateTimeImmutable::createFromFormat('d/m/Y', $date)) {
+            throw new InvalidArgumentException('The date must be in DD/MM/YYYY format.');
+        }
 
-    /**
-     * This method wait an ISO 3166-1 alpha.
-     */
-    public function destIso(string $isoCode)
-    {
-        $this->options['dest_iso'] = self::validateDestIso($isoCode);
+        $this->options['date'] = $date;
 
         return $this;
     }
 
-    /**
-     * This method will automatically set personnalise = 1 (according to SMSBOX documentation).
-     */
-    public function variable(array $variable)
+    public function hour(string $hour): self
     {
-        $this->options['variable'] = $variable;
+        if (isset($this->options['dateTime'])) {
+            throw new InvalidArgumentException(sprintf('Either %1$s::dateTime() or %1$s::date() and %1$s::hour() must be called, but not both.', self::class));
+        }
 
-        return $this;
-    }
+        if (false === \DateTimeImmutable::createFromFormat('H:i', $hour)) {
```

Comment:
same everywhere similar:
```suggestion
        if (!\DateTimeImmutable::createFromFormat('H:i', $hour)) {
```